x/mongo/driver/compression.go

```diff
@@ -26,48 +26,70 @@ type CompressionOpts struct {
 	UncompressedSize int32
 }
 
-var zstdEncoders sync.Map // map[zstd.EncoderLevel]*zstd.Encoder
+func zstdNewWriter(lvl zstd.EncoderLevel) *zstd.Encoder {
```

Comment:
I think renaming `zstdNewWriter` to `mustZstdNewWriter` should make it clearer that this will panic and follows the general naming convention of Go functions that will panic.

The issue with putting this in an `init` function is that it the order in which `init` function calls is a bit tricky (though now formalized with go1.21) this can make it dangerous for any other `init` to use any of the code here that relies on the `zstdEncoders`. 

I'm also of the opinion that without a really really strong use-case packages should **never** have an `init` function (at Lyft I banned the use of `init` functions in libraries unless there was an excellent case for it because of the trouble they caused  - ran Go development there).

If we really wanted to doe this without exposing this function we could use this beast, but IMHO it's not 
```go
var zstdEncoders = func() [zstd.SpeedBestCompression + 1]*zstd.Encoder {
	f := func(lvl zstd.EncoderLevel) *zstd.Encoder {
		enc, err := zstd.NewWriter(nil, zstd.WithEncoderLevel(lvl))
		if err != nil {
			panic(err) // this should never happen
		}
		return enc
	}
	return [zstd.SpeedBestCompression + 1]*zstd.Encoder{
		0:                           nil, // zstd.speedNotSet
		zstd.SpeedFastest:           f(zstd.SpeedFastest),
		zstd.SpeedDefault:           f(zstd.SpeedDefault),
		zstd.SpeedBetterCompression: f(zstd.SpeedBetterCompression),
		zstd.SpeedBestCompression:   f(zstd.SpeedBestCompression),
	}
}
```

Reply:
Renaming it to `mustZstdNewWriter` sounds like a great idea!